src/WingetCreateCLI/Program.cs

```
@@ -1,88 +1,104 @@
     {
-        private static async Task<int> Main(string[] args)
+        private static async Task<int> Main(string[] args)
         {
             Logger.Initialize();
             UserSettings.FirstRunTelemetryConsent();
             TelemetryEventListener.EventListener.IsTelemetryEnabled();
-
-            string arguments = string.Join(' ', Environment.GetCommandLineArgs());
-            Logger.Trace($"Command line args: {arguments}");
-
-            Parser myParser = new Parser(config => config.HelpWriter = null);
+
+            string arguments = string.Join(' ', Environment.GetCommandLineArgs());
+            Logger.Trace($"Command line args: {arguments}");
+
+            Parser myParser = new Parser(config => config.HelpWriter = null);
 
             var types = new Type[] { typeof(NewCommand), typeof(UpdateCommand), typeof(SubmitCommand), typeof(SettingsCommand), typeof(TokenCommand), typeof(CacheCommand) };
             var parserResult = myParser.ParseArguments(args, types);
 
             BaseCommand command = parserResult.MapResult(c => c as BaseCommand, err => null);
 
-            if (command == null)
+            string token = command?.GitHubToken ?? GitHubOAuth.ReadTokenCache();
+            try
+            {
+                string latestVersion = await GitHub.GetLatestRelease(token);
+                if (latestVersion.TrimStart('v').Split('-').First() != Utils.GetEntryAssemblyVersion())
+                {
+                    Logger.WarnLocalized(nameof(Resources.OutdatedVersionNotice_Message));
+                    Logger.WarnLocalized(nameof(Resources.GetLatestVersion_Message), latestVersion, "https://github.com/microsoft/winget-create/releases");
+                    Logger.WarnLocalized(nameof(Resources.UpgradeUsingWinget_Message));
+                    Console.WriteLine();
+                }
+            }
+            catch (Exception ex) when (ex is Octokit.ApiException || ex is Octokit.RateLimitExceededException)
+            {

```

Add a comment in the block to explain why we're ignoring the exception

---

Added comment to block explaining why we're ignoring exception